lxd/backup.go

```diff
@@ -399,3 +384,76 @@ func backupCreateTarball(s *state.State, path string, backup backup) error {
 
 	return nil
 }
+
+func pruneExpiredContainerBackupsTask(d *Daemon) (task.Func, task.Schedule) {
+	f := func(ctx context.Context) {
+		pruneExpiredContainerBackups(ctx, d)
+	}
+
+	pruneExpiredContainerBackups(context.Background(), d)
+
+	first := true
+	schedule := func() (time.Duration, error) {
+		interval := 24 * time.Hour
+
+		if first {
+			first = false
+			return interval, task.ErrSkip
+		}
+
+		return interval, nil
+	}
+
+	return f, schedule
+}
+
+func pruneExpiredContainerBackups(ctx context.Context, d *Daemon) {
```

Comment:
So this approach is probably fine for now, but I think we'll soon want a generic way of waiting until a particular time while allowing to reset that target time.

So effectively for both backup and snapshot expiry, we'd have a goroutine which waits until the next one expires, when that hits, it gets deleted and then waits until the one after that expires, ...
The added complexity there is that we'll need a way to reset that timer whenever a new snapshot or backup is created, if it expires BEFORE the current target time.

Reply:
Sounds good to me.

Reply:
This chunk needs porting to using a background operation as I've done a number of others yesterday, I'll send a commit which does that.